Add citation guidelines to the manual

ckhroulev opened the pull request:
This PR adds citation guidelines to the manual, including information about preserving data needed to aid reproducibility.

The goals are: _improve citation guidelines_ and _reduce duplication_.

Until now we had some recommendations
- in `README.rst`
- in `ACKNOWLEDGE.rst`
- in `CITATION.cff`
- at https://www.pism.io/citing/

plus some seemingly related text in `doc/sphinx/acknowledgments.rst`.

Now all citation guidelines are in `doc/sphinx/citing/index.rst` (named this way to be available at `www.pism.io/docs/citing` once this version of the manual is online).

- The "Citing PISM" menu item on `pism.io` will redirect to `www.pism.io/docs/citing`. The webpage at https://www.pism.io/citing/ can be removed.
- `CITATION.cff` and `README.rst` are updated and contain messages sending users to `www.pism.io/docs/citing` or its local (plain text) alternative.
- `ACKNOWLEDGE.rst` is removed since it is no longer needed.
- `doc/sphinx/acknowledgments.rst` should not be an issue any more (now citation guidelines are easy to find).

**Checklist**

- [x] update documentation
- [ ] update [`CHANGES.rst`](CHANGES.rst)
- [x] ~add regression tests~

ckhroulev commented:
@gassmoeller @ljhwang I added more detailed PISM citation guidelines and attempted to reduce duplication. Please let me know what you think -- any input is much appreciated. Thanks!

ljhwang commented:
This looks very thorough! Thanks.

A few comments:

- Under the section "Studies using customized versions of PISM", I am wondering if there is a statement, if applicable, that strongly encourages modifications to be contributed back  to the repository.

- I have seen that some people make a separate repo for the publication which I presume is frozen.  However, in some cases  they might cite their fork  of the code but continue to develop on this copy.  In that case, it may be better that the citation include the git short hash.

ckhroulev commented:
@ljhwang Thanks for the feedback!

I added a statement reminding users to consider contributing their modifications.

I agree that *if* one uses a publicly available repository to store the code then a citation should include the short hash of the relevant commit. I am reluctant to recommend this, though: I think it is always better to archive the code and get a DOI, especially since it's easy and free.

PS: Based on the feedback I got from a PISM contributor I split recommendations into two sections: "Citing PISM" and "Publishing" with hyperlinks from one to the other. (Rationale: publishing modeling results involves more than citing PISM and citing PISM may be needed when publishing something else, i.e. not PISM results.)
